EnhancedMediaPlayer/MediaPlayerControllerRepresentable.swift

```diff
@@ -0,0 +1,70 @@
+//  MediaPlayerControllerRepresentable.swift
+
+import AVKit
+import SwiftUI
+
+public struct MediaPlayerControllerRepresentable: UIViewControllerRepresentable {
+    @ObservedObject private var playerManager: PlayerManager
+    private let mediaURL: URL
+
+    init(mediaURL: URL) {
+        self.mediaURL = mediaURL
+        self.playerManager = .init(playerController: .init())
+    }
+
+    public func makeUIViewController(context: Context) -> AVPlayerViewController {
+        setupPlayerController(playerManager.playerController)
+
+        return playerManager.playerController
+    }
+
+    public func updateUIViewController(_ playerController: AVPlayerViewController, context: Context) {
+        if let currentURL = playerController.player?.currentItem?.asset as? AVURLAsset,
+           currentURL.url != mediaURL {
+            playerManager.playerController = playerController
```

Comment:
yes, whenever `updateUIViewController` runs, when there's an update to the controller, it gets a new instance of `playerController`. If I don't update the `playerManager`, it will be out of sync when, for example, there's a change in the media url. The action buttons will stop working for the new media, since they rely on the manager